### ISSUE TITLE
Error in utils::combn(p, m = 2, FUN = prod, na.rm = TRUE): n < m in `paRao()`

### ISSUE THREAD

**Issue by Nowosad** [mattmar/rasterdiv#8]
This seems to be some kind of a regression -- the below code worked for the previous version of the package (as you also may see at https://jakubnowosad.com/posts/2024-10-20-spatcomp-bp2/#the-difference-between-a-focal-measure-of-two-rasters):

``` r
library(terra)
library(rasterdiv)
ndvi2018_tartu = rast("https://github.com/Nowosad/comparing-spatial-patterns-2024/raw/refs/heads/main/data/ndvi2018_tartu.tif")
ndvi2018_tartu_int = as.int(ndvi2018_tartu * 100)
ndvi2018_tartu_rao = paRao(ndvi2018_tartu_int, window = 5, progBar = FALSE)
#> Warning in paRao(ndvi2018_tartu_int, window = 5, progBar = FALSE): Simplify=0.
#> Rounding data to 0 decimal places.
#> 
#> 
#> Processing alpha: 1 Moving Window: 5
#> 
#> 
#> Processing alpha: 1 Moving Window: 5
#> Error in utils::combn(p, m = 2, FUN = prod, na.rm = TRUE): n < m
```

**Comment by mattmar** [comment]
This happens because you only have 1 pixel value in one of the moving windows due to NAs on the image edge and Rao's index is undefined in this case; to overcome the error set na.tolerance=0 and that window will be set to NA. I will add a warning message at the first chance.

**Comment by mattmar** [comment]
Patched in rasterdiv 0.3.6. Window with number of NA < na.tolerance but with only 1 valid category are now set to 0.

**Comment by Nowosad** [comment]
Thanks @mattmar. 

One small additional question -- why the "Processing alpha: 1 Moving Window: 5" message is shown twice when running the calculations:

```r
> ndvi2018_tartu_rao = paRao(ndvi2018_tartu_int, window = 5, progBar = FALSE)


Processing alpha: 1 Moving Window: 5


Processing alpha: 1 Moving Window: 5
```
